src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1443,8 +1451,20 @@ private async ValueTask NotifyFeaturesAboutStateChangeAsync(DatabaseRecord recor
                         InitializeFromDatabaseRecord(record);
                         IndexStore.HandleDatabaseRecordChange(record, index);
                         ReplicationLoader?.HandleDatabaseRecordChange(record, index);
-                        EtlLoader?.HandleDatabaseRecordChange(record);
-                        SubscriptionStorage?.HandleDatabaseRecordChange(record.Topology);
+
+                        // ReSharper disable once MethodSupportsCancellation
+                        await _updateValuesLocker.WaitAsync();
```

Comment:
you take both locks here? this is a recipe for a deadlock

Reply:
It is strange for me that we need this lock at the first place.
I understand that we have `HandleDatabaseRecordChange` in the `ValueChanged` code path, which seems to be wrong?

Reply:
There are 3 areas that we update.
- Backup
- ETL
- Subscriptions

The subscription update has the same code path in `RecordChanged` and `ValueChanged`.
The other 2 are different.

Reply:
I would split it into 2 methods `HandleDatabaseRecordChange` and `HandleValueChange` and not mix the locks

Reply:
+1

Reply:
Talked with @karmeli87.
Separating `HandleDatabaseRecordChange` and `HandleValueChange` would necessitate substantial modifications. We must ensure that the locks are acquired and disposed of in the correct sequence.